scripts/images_learn_sha.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+images_amd64=${IMAGES_AMD64}
+images_arm64=${IMAGES_ARM64}
+CRANE=${CRANE}
+
+if [ -z "$images_amd64" ]; then
+    echo "IMAGES_AMD64 is not set"
+    exit 1
+fi
+
+if [ -z "$images_arm64" ]; then
+    echo "IMAGES_ARM64 is not set"
+    exit 1
+fi
+
+if [ -z "$CRANE" ]; then
+    echo "CRANE is not set"
+    exit 1
+fi
+
+learn_data=()
+
+for image in $images_amd64; do
+    image_no_digest=$(echo -n "$image" | cut -d@ -f1)
+    find=$(echo -n "$image" | cut -d@ -f2)
+    replace=$($CRANE digest --platform "linux/amd64" "$image_no_digest")
+
+    if [ "$find" == "$replace" ]; then
+        continue
+    fi
+
+    learn_data+=("s|$find|$replace|g\n")
+done
+
+for image in $images_arm64; do
+    image_no_digest=$(echo -n "$image" | cut -d@ -f1)
+    find=$(echo -n "$image" | cut -d@ -f2)
+    replace=$($CRANE digest --platform "linux/arm64" "$image_no_digest")
+
+    if [ "$find" == "$replace" ]; then
+        continue
+    fi
+
+    learn_data+=("s|$find|$replace|g\n")
+done
```

Comment:
comment: this could be a function, but that's complicated enough in bash that it's fine to copy here I think!